### ISSUE TITLE
Partial opacity fails silently on a group that is much larger than the view

### ISSUE THREAD

**Issue by RickMohr** [paperjs/paper.js#2001]
In [this sketch](http://sketch.paperjs.org/#V/0.12.15/S/TZAxT8QwDIX/itXlWilEBYnlENMNrEiwtR3S1OlFV+wqSTlB1f9O0sAJL078Je89eS1IfWBxLN4uGPS5EIXmId3NQjpYJujtWBo7TSee2Anw9hsFsDEeQwVrSxBLM/kAIzwD4RVeHC9zWT1lZthBmR/M7O2uyQaaJmsIqDsBTZbdj3U26bqbfKok/KrCWZ6s0xOWKzg12MUf4b6OP/4lvLlslVTD8M7l+Jdly22UPCttw1cMLB8ji/OWPi1eZXCK/KQClrOa0cl9qJECuiTSUlrHoZ8WPIjknLzvYk9wR6NDpMgeMvtFca+9Q3WZ2VLwxbHpth8=) the function `big` creates a group with 3 circles and sets its opacity to 0.5. In the first (blue) group the circles are separated by 10,000 units, and the first circle is correctly seen at half opacity. But in the second (green) group the circles are separated by 20,000 units, and the first circle is not displayed. (If you change the size to 10,000 it will be displayed.)

This is on Windows/Chrome. On my iPhone the problem starts at a much smaller number -- 1100 units works but 1200 doesn't. Looking at the code I see that drawing an item with partial opacity uses an offscreen canvas, which likely is size-limited and platform-dependent.

In my actual application I was able to work around the problem by setting `visible` to `false` for items completely outside the view. But  I could have saved a couple days of work if a warning was logged to the console in this situation.

**Comment by lehni** [comment]
Yeah certain operations force the whole group / layer to be rasterized into a separate canvas. Since your units are huge, this means a massive buffer would nee to be created, and I guess you're simply running into a OOM exception here. I'd recommend using the alpha channel on the fill / stroke color instead, if this is an option?

**Comment by RickMohr** [comment]
Thanks for the suggestion. I ended up clipping to the view bounds, rasterizing, and changing opacity on the (now reasonably-sized) raster. Thanks for the simplicity of `Item.rasterize()`!

Leaving this open in case it's possible to issue a warning when the offscreen canvas can't be allocated.
